modules/core/keeper/msg_server.go

```diff
@@ -776,16 +775,11 @@ func (k Keeper) ChannelUpgradeTry(goCtx context.Context, msg *channeltypes.MsgCh
 		return nil, errorsmod.Wrap(err, "channel upgrade try failed")
```

Comment:
Just for my understanding: if we return an error here, the error receipt that is written on the statement above is not reverted?

Reply:
The error receipt is only written if `channeltypes.IsUpgradeError(err)`, this is a different if branch. 

The one above where the error receipt is written returns: 
`return &channeltypes.MsgChannelUpgradeTryResponse{Result: channeltypes.FAILURE}, nil`